### PR TITLE
Policy module fix for modify with template name

### PR DESCRIPTION
**Pull request by mmudigon**
This PR fixes a couple of issues in dcnm_policy module. (https://github.com/CiscoDevNet/ansible-dcnm/issues/60)

1. When a policy is created (state = merged) using a template name that is already used by another policy, the new policy is not created. Because the existing code allows modifications only with policy-ids. This PR updates the code to fix this which will now allow the new policy to be created instead of treating it like a modify operation.
2. When a policy is deleted, all the data from DCNM server is deleted appropriately. But the configuration related to the policy is not removed from the switch. This PR fixes an issue with the mark-delete path which has changed in 11.5. With this fix the policy configuration is removed from the switch.

### CODE REVIEW

File: tests/integration/targets/dcnm_policy/tests/dcnm/dcnm_policy_delete.yaml [hunk 1]
```diff
@@ -106,28 +106,7 @@
         that:
           - 'item["RETURN_CODE"] == 200'
           - 'item["MESSAGE"] == "OK"'
-      when: (my_idx < (result["diff"][0]["deleted"] | length))
```
Comment (mikewiebe) [inline tests/integration/targets/dcnm_policy/tests/dcnm/dcnm_policy_delete.yaml:109]: Can you add test cases to explicitly cover the following:

* Create several policies with a template1 (no variables)
* Add new policies using template1 where `description` and `priority` are different then existing policies
* Force creation of new policies with the force flag set to true
* Verify policies are not created when the force flag is false and policies match already existing policies
* Repeat this for template2 (with variables)

Comment (mikewiebe) [reply tests/integration/targets/dcnm_policy/tests/dcnm/dcnm_policy_delete.yaml:109]: Some of these might already be covered but want to make sure we have coverage for all of them

Comment (mmudigon) [reply tests/integration/targets/dcnm_policy/tests/dcnm/dcnm_policy_delete.yaml:109]: Mike these are already covered as part of some of the other cases. Anyway I added some more cases to explicitly cover the new cases. Pushed the changes again.

**Comment by mikewiebe** [comment]
@mmudigon This is looking much better but I found a couple of issues:

1. I can't figure out how to delete a single policy.  When I specify a policy name it still deletes all of the policies associated with the template.
    * IMO we should be able to delete a single policy or targeted set of policies when the name is the name of the policy id
    * Also, since the default behavior seems to be to delete all policies, we should be able to omit the `config:` keyword
2. When the `deploy` flag is set to true I see the `changed` flag being set even if no new policies are being created

**Pull request merged**